src/main/java/com/easytestit/generatexml/service/ReportService.java

```diff
@@ -0,0 +1,486 @@
+    private int countScenariosInSuite = 0;
+    private Long durationOfTest = 0L;
+    private Long durationOfAllTest = 0L;
+    private Long durationOfAllTestFromAllSuites = 0L;
+
+    /**
+     * The main goal in this method is to convert JAVA object which was deserialized from JSON file
+     * to other JAVA object which will serialize to XML file
+     * Here concentrate logic to calculate and handle data and put this data in DTO
+     *
+     * @param features prepared JAVA feature class which should be convert to another DTO
+     * @return prepared DTO aggregated class with all needed data for serialize it to XML file
+     */
+    public XMLReport transformFeaturesToReport(final Collection<Feature> features) {
+        if (features != null) {
+            Collection<SingleReportSuite> singleReportSuites = new ArrayList<>();
+
+            if (features.isEmpty()) {
+                throw new GenerateXMLReportException("Empty argument provided");
+            }
+
+            features.forEach((feature) -> {
+                if (feature != null) {
+                    if (feature.getTags() != null) this.tagProcessing("featureTags", feature.getTags());
+                    if (feature.getElements() != null)
+                        feature.getElements().forEach((element) -> this.elementProcessing(feature, element));
+
+                    singleReportSuites.add(getSingleReportSuite(feature, features));
+                    durationOfAllTestFromAllSuites = this.durationOfAllTestFromAllSuites + this.durationOfAllTest;
+                    resetVariables();
+                } else {
+                    throw new GenerateXMLReportException("Argument feature from the Array features should not be null but is null");
+                }
+            });
+
+            return getReportSuites(countFailures, countScenarios, durationOfAllTestFromAllSuites, singleReportSuites);
+        } else {
+            throw new GenerateXMLReportException("Argument features should not be null but is null. See detailed stack trace: ", new NullPointerException());
+        }
+    }
+
+    /**
+     * Separate method for better handling code
+     * Created for getting data about tags and pass these tags to other function
+     *
+     * @param tagKey the key value for processing tags for specific block of feature file
+     * @param tags   object which should be handled
+     */
+    private void tagProcessing(final String tagKey, final Collection<Tag> tags) {
+        if (tags != null & tagKey != null) {
+            tags.stream()
+                    .filter(t -> !featureTags.contains(t.getName()))
+                    .forEach(t -> tagsValue += t.getName().concat(" "));
+            if (!tagsValue.isEmpty()) featureTagsMap.put(tagKey, tagsValue);
+            tagsValue = "";
+        } else {
+            throw new GenerateXMLReportException("Arguments tags, tagKey should not be null but are null. See detailed stack trace: ", new NullPointerException());
```

Comment:
Same as above

Reply:
Done. Will be available after pushing a new commit.